src/v/datalake/translation/deps.cc

```diff
@@ -181,9 +181,22 @@ class partition_data_source : public data_source {
     ss::future<kafka::offset> wait_for_data_to_translate(
       std::optional<kafka::offset> last_translated_offset,
       ss::abort_source& as) final {
-        // todo: add logic to wait for enough data to translate.
-        // currently we just break even if a single batch of data is available
-        while (!has_more_data_to_translate(last_translated_offset)) {
+        // naively wait a few polling periods after some data appears, hopefully
```

Comment:
I'm wondering how problematic this is in practice and how much it makes the difference, compared to e.g. improving the scheduling policy. Conceptually it feels like if there is even a single batch, there's no harm in telling the scheduler about it as long as the scheduler prioritizes appropriately w.r.t size, deadlines, etc

Reply:
yeah that's a fair point. we're still going to wait to be scheduled in, so by the time we wake up there might be more data there anyway. @bharathv does the default scheduling policy take that into account?

Reply:
https://github.com/redpanda-data/redpanda/pull/25152#discussion_r1976681961